src/Riok.Mapperly/MapperGenerator.cs

```diff
@@ -22,39 +22,63 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
         var mapperClassDeclarations = SyntaxProvider.GetClassDeclarations(context);
 
         var compilationAndMappers = context.CompilationProvider.Combine(mapperClassDeclarations.Collect());
-        context.RegisterImplementationSourceOutput(compilationAndMappers, static (spc, source) => Execute(source.Left, source.Right, spc));
+        var mappersWithErrors = compilationAndMappers.Select((x, token) => BuildDescriptors(x.Left, x.Right, token));
```

Comment:
Would rename `token` to `ct` or `cancellationToken` as only token seems more syntax related to me in this context.